### PR TITLE
[AR-3547] Fix null 'id' field

### PR DESCRIPTION
**Pull request by jwliechty**
[AR-3547]

[AR-3547]: https://indigobio.atlassian.net/browse/AR-3547

The logatron log lines were printing the id field just fine when rails
was started via the command line. However, when ran in docker, the id
field was `null` each time logatron logged a line. After some
investigation, it was found that it had to do with the way unicorn and
logatron are loaded before rails when running in docker vs logatron being
loaded when within the rails context when starting on the command line.
To fix the problem, the logatron railtie must be included in the
`config/application.rb` after `Rails` is present but before bundler
requires all the things. The logatron railtie will insert its
`config.after_initialize` before the lograge railtie is loaded. This will
ensure the `config.after_initialize` block in the logatron railtie will
run before the one specified in lograge. The lograge
`config.after_initialize` block will then be guaranteed to have both
logatron and any lograge configuration complete before running.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -27,26 +27,39 @@ The goal: constrain all configuration regarding the logging to go through
 logatron and not the internals of logatron (such as lograge). Consumers of
 logatron should not have to know the internals.
 
-Logatron must be configured before anything else since there can be other
-configuration dependent upon those values (lograge for sure). This is done by
-doing that configuration in an initializer file that will be listed first when
-sorted in that directory alphanumerically. Putting an underscore (`_`) in front
-of the name typically will move it to the top of the list -
-`config/initializers/_log.rb`. Here is an example configuration.
+The logatron railtie `after_initialize` must run before the `after_initialize`
```
Comment (wintonpc) [inline README.md:30]: why?

File: README.md [hunk 2]
```diff
@@ -27,26 +27,39 @@ The goal: constrain all configuration regarding the logging to go through
...
+of the lograge railtie. This means that the logatron railtie must be required
+before lograge. Do not require lograrage directly in rails. Instead add the
```
Comment (wintonpc) [inline README.md]: typo: lograrage

File: README.md [hunk 3]
```diff
@@ -27,26 +27,39 @@ The goal: constrain all configuration regarding the logging to go through
...
+following line to `config/application.rb` somewhere after rails has been
+required but before bundler requires all the things.
```
Comment (wintonpc) [inline README.md]: what does "all the things" mean?

File: lib/logatron/railtie.rb [hunk 4]
```diff
@@ -95,7 +79,25 @@ class Railtie < Rails::Railtie
...
+        'The config.after_initialize of Logatron must run before that ',
+        'of Lograge::Railtie.'
+    ].join
+end
+begin
+  require 'lograge'
+rescue LoadError
+  msg = [
+      'Include "lograge" in your bundle. It was left out of the gemspec of',
+      'Logatron to keep dependencies down for projects that do not use Rails.'
```
Comment (wintonpc) [inline lib/logatron/railtie.rb:100]: I assume the problem dependency chain starts with `lograge` -> `railties` ?

**Comment by wintonpc** [comment]
how did you verify this? or what is the verification plan?

**Review by wintonpc — CHANGES_REQUESTED** [review]

**Comment by wintonpc** [comment]
@jwliechty and I agreed that a task should be entered for eliminating the dependency on `lograge` to avoid this madness.

**Comment by jwliechty** [comment]
@wintonpc I have objective evidence I attached to the JIRA case.

File: lib/logatron/railtie.rb (hunk 4 above)
Comment (jwliechty) [reply lib/logatron/railtie.rb:100]: I am uncertain what you mean by this.

Comment (wintonpc) [reply lib/logatron/railtie.rb:100]: don't worry about it.

**Review by wintonpc — APPROVED** [review]

**Comment by jwliechty** [comment]
Made case https://indigobio.atlassian.net/browse/AR-3512

**Pull request merged**